new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiMultiModel.java

```diff
@@ -0,0 +1,332 @@
+    // model files are in the domain configuration model's ConfigMap.
+    // In addition, the first model defines a second DataSource "TestDataSource3", which should
+    // also be in the resultant configuration.
+    final String modelFileName1 = "multi-model-two-ds.10.yaml";
+    final String modelFileName2 = "multi-model-one-ds.20.yaml";
+
+    final String dsName = "TestDataSource";
+    final String dsName3 = "TestDataSource3";
+
+    final String configMapName = "ds-multi-model-cm";
+
+    logger.info("Create the repo secret {0} to pull the image", REPO_SECRET_NAME);
+    assertDoesNotThrow(() -> createDockerRegistrySecret(domainNamespace),
+            String.format("createSecret failed for %s", REPO_SECRET_NAME));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(
+        adminSecretName,
+        domainNamespace,
+        ADMIN_USERNAME_DEFAULT,
+        ADMIN_PASSWORD_DEFAULT),
+        String.format("createSecret failed for %s", adminSecretName));
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    String encryptionSecretName = "encryptionsecret";
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(
+        encryptionSecretName,
+        domainNamespace,
+        "weblogicenc",
+        "weblogicenc"),
+        String.format("createSecret failed for %s", encryptionSecretName));
+
+    logger.info("Create ConfigMap {0} that contains model files {1} and {2}",
+        configMapName, modelFileName1, modelFileName2);
+
+    Map<String, String> data = new HashMap<>();
+    addModelFile(data, modelFileName1);
+    addModelFile(data, modelFileName2);
+
+    createConfigMapAndVerify(configMapName, domainUid, domainNamespace, data);
+
+    logger.info("Create the domain resource {0} in namespace {1} with ConfigMap {2}",
+        domainUid, domainNamespace, configMapName);
+    Domain domain = createDomainResource(domainUid, domainNamespace, adminSecretName,
+        REPO_SECRET_NAME, encryptionSecretName, replicaCount, configMapName);
+    
+    createDomainAndVerify(domain, domainNamespace);
+
+    // check admin server pod is ready
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+
+    // check managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server pod {0} to be ready in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
```

Comment:
This check has to be done right after the checkPodReady(adminServerPodName, domainUid, domainNamespace);
I have seen that when admin services are  not created because of any errors like in T3ChannelPort not matching with nodePort, the managed servers pods are not created. But it throws off us to the debugging as to why the managed servers are not created than concentrating on the admin server services creation failures.

Reply:
There are some explanations in the beginning of the test code where we specify the variables for the two model files. I just added more explanations. Can you check and see if it is clearer to you?

Reply:
changed the order.